src/averageAges.js

```diff
@@ -15,11 +15,22 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const menLeng = (people
+    .filter((i) => arguments.length < 2
+      ? i.sex === 'm'
+      : i.sex === 'm' && Math.ceil(i.died / 100) === century))
+    .length;
+
+  const age = +(people
+    .filter((i) => arguments.length < 2
+      ? i.sex === 'm'
+      : i.sex === 'm' && Math.ceil(i.died / 100) === century)
```

Comment:
there is similar logic here, you can combine it together to find all men first and then use this variable to get the length and calculate the sum